### ISSUE TITLE
Add Soil Type Classifier using CNN Project

### ISSUE THREAD

**Issue by keshripritesh** [omroy07/AgriTech#392]
<h3>🧩 <strong>Issue: Add Soil Type Classifier using CNN Project to GSSoC'25 Repository</strong></h3>
<hr>
<h3>🌱 <strong>Project Title</strong></h3>
<p><strong>Soil Type Classifier using CNN | Agriculture AI</strong></p>
<hr>
<h3>🧠 <strong>Project Description</strong></h3>
<p>This project introduces a <strong>deep learning-based image classification system</strong> that automatically identifies <strong>different soil types</strong> using raw images. By leveraging Convolutional Neural Networks (CNNs), the model assists <strong>farmers, agronomists, and agricultural researchers</strong> in making faster, data-driven decisions about <strong>crop selection, irrigation planning, and soil treatment</strong>.</p>
<p>This project is a part of the <strong>Agriculture AI initiative under GirlScript Summer of Code (GSSoC’25)</strong>.</p>
<hr>
<h3>🧾 <strong>Problem Statement</strong></h3>
<p>Manual identification of soil types is a <strong>time-consuming</strong> and <strong>expert-dependent</strong> process. Automating this task with computer vision can significantly improve <strong>agricultural productivity and decision-making</strong> by providing quick and reliable soil classification insights.</p>
<hr>
<h3>🌍 <strong>Objective</strong></h3>
<p>Develop a <strong>CNN-based image classifier</strong> capable of recognizing and categorizing soil samples into <strong>five major soil types</strong> based on texture, color, and composition features extracted from images.</p>
<hr>
<h3>🧩 <strong>Supported Soil Categories</strong></h3>
<ul>
<li>
<p>🟫 <strong>Black Soil</strong></p>
</li>
<li>
<p>🌋 <strong>Cinder Soil</strong></p>
</li>
<li>
<p>🪨 <strong>Laterite Soil</strong></p>
</li>
<li>
<p>🌿 <strong>Peat Soil</strong></p>
</li>
<li>
<p>🟡 <strong>Yellow Soil</strong></p>
</li>
</ul>
<hr>
<h3>⚙️ <strong>Tech Stack</strong></h3>

Component | Technology Used
-- | --
Language | Python
Framework | TensorFlow / Keras
Model Type | Convolutional Neural Network (CNN)
Visualization | Matplotlib
Utilities | NumPy, PIL

<h3>📸 <strong>Output Section (Before &amp; After Enhancement)</strong></h3>
<p><strong>Before Enhancement:</strong><br>
Epoch 10/20
Training Accuracy: 72.45%
Validation Accuracy: 68.90%
Training Loss: 0.65
Validation Loss: 0.72

<p><strong>After Enhancement:</strong><br>
Epoch 25/25
Training Accuracy: 95.87%
Validation Accuracy: 92.14%
Training Loss: 0.11
Validation Loss: 0.18

<hr>
<h3>🧑‍💻 <strong>Contributor</strong></h3>
<p><strong>Pritesh  Keshri</strong></p>
<hr>
<h3>🔢 <strong>Issue Number</strong></h3>
<p><em>(To be filled by project maintainers)</em></p>
<hr>
